src/components/AdvertiserName/FollowUserButton/FollowUserButton.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { ErrorModal } from '@/components/Modals';
+import { OnboardingTooltip } from '@/components/OnboardingTooltip';
+import { BUY_SELL_URL } from '@/constants';
+import { api } from '@/hooks';
+import { useAdvertiserStats, useIsAdvertiserBarred, useModalManager } from '@/hooks/custom-hooks';
+import { getInvalidIDErrorMessage, getInvalidIDTitle } from '@/utils';
+import { StandaloneUserPlusFillIcon } from '@deriv/quill-icons';
+import { Localize, useTranslations } from '@deriv-com/translations';
+import { Button, useDevice } from '@deriv-com/ui';
+import { LocalStorageConstants } from '@deriv-com/utils';
+
+type TFollowUserButtonProps = {
+    id?: string;
```

Comment:
is this the advertiser id? can we require this prop?

Reply:
I typed it like this because advertiserStats.id can be undefined, but i don't think this'll cause a massive issue here since we're only using this FollowButton here
![Screenshot 2024-10-09 at 2 17 44 PM](https://github.com/user-attachments/assets/157c6004-211e-4611-8971-a9a1e8ca8a1b)